### PR TITLE
Login ui for the website

### LINKED ISSUE #16: Create Login UI
## Description
As a user, I need to be able to login to OSP.

## Mocks
As finalized in #3 

### Update [Required]
- [ ] Create Login UI
- [ ] Throw validation errors on the UI if the username is invalid or if the username and password do not match or if any of the mandatory fields are missing.
- [ ] If login is successful, then I should be redirected to the dashboard
- [ ] If login fails, then I should receive proper feedback
- [ ] store authentication token in local storage of browser

 must have the fields for login:
- [ ] username
- [ ] password(should be masked)
- [ ]  must have Register and Login buttons

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
2 days

### PR DESCRIPTION
**Pull request by bismitaguha**
### Description

- Modified login action accordingly
- Added Login Component
- Added routing for Login

Fixes #16 

### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Currently requires double click to enter Dashboard Page on successful Login

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by bismitaguha** [comment]
@abha224 @sidvenu 
Errors arising:
- As registration needs confirmation I have removed any redirect from that component.
- Login happens successfully on one click, but redirect to dashboard is made on second click.
- Whenever Login form is submitted a warning is generated: `Form submission canceled because the form is not connected`

Please clarify these!

**Comment by sidvenu** [comment]
@bismitaguha the issues are related to each other. When a button is clicked, browsers could interpret it differently to be a submit button (whereas it is a button to run a piece of code, and then use code to navigate away), and thus your ```onClick``` and the browser is competing with each other to submit the form.

Try adding ```type="button"``` to the button attributes or ```evt.preventDefault()``` in the button's ```onClick``` function.

**Comment by sidvenu** [comment]
As we talked about in the weekly session, please do try to give separate PRs for each issue. It's hard to go through 22 files with about 1000 LoC additions. Within a single PR, do break it down into logically separate commits.

And please follow the commit message style guide: https://github.com/anitab-org/mentorship-android/wiki/Commit-Message-Style-Guide. Is it possible for you to change the commit messages to follow the style guide?

**Review by sidvenu — CHANGES_REQUESTED** [review]
Please do change the commit messages as per the comment above.

**Comment by bismitaguha** [comment]
@sidvenu I was suggested by @Monal5031  to proceed to further issues and rebase later before merging. So I am doing the same. I hope it's fine.

**Comment by sidvenu** [comment]
@bismitaguha If that is the case, sure, we can use the message style guide while merging. But please do keep the above suggestions in mind for your next PRs :smile:.

**Comment by bismitaguha** [comment]
> @bismitaguha If that is the case, sure, we can use the message style guide while merging. But please do keep the above suggestions in mind for your next PRs .

I have made separate PRs based on your suggestions in the meeting. I will look into the commit messages. 😊

**Comment by sidvenu** [comment]
Amazing. Let us know if the above solution to the ```Form submission canceled because the form is not connected``` error is working.

**Comment by Monal5031** [comment]
@bismitaguha I think there was a miscommunication, I suggested to start working on a separate branch so that would help you save time while the main PR is in review. I didn't meant to add the changes in same PR

**Comment by bismitaguha** [comment]
This is in a separate branch. The previous PR [#27](https://github.com/anitab-org/open-source-programs-web/pull/27). Once the previous one gets merged I will rebase this. I did that same with the previous issue . It was in backend repo. I hope that's fine?? @Monal5031

**Comment by Monal5031** [comment]
Ahh okay, by @sidvenu 's comments I thought you made a single PR for different issues. It's okay I guess if the PRs are for different issues just make sure to mark them as WIP with a label or in title.

@sidvenu @abha224 Please let us know if this strategy to save time is okay with you all?

**Comment by sidvenu** [comment]
@Monal5031 if two issues are heavily dependent on each other that it'll hinder developement of one without the other, then absolutely - we can have a single PR for both the issues.

If that is not the case, I feel separate PRs would be ideal - since it is just good open source practice and it prevents big PRs to a certain extent. What do you think?

**Comment by Monal5031** [comment]
Agreed @sidvenu makes total sense

**Comment by abha224** [comment]
> * Login happens successfully on one click, but redirect to dashboard is made on second click.
> * Whenever Login form is submitted a warning is generated: `Form submission canceled because the form is not connected`

@bismitaguha Im not able to reproduce the error? can you provide the steps again?

File: src/components/Login.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Login extends Component {
     })
 
     submitLogin = () => {
-
+        console.log("hello")
```
Comment (abha224) [inline src/components/Login.js]: remove this line

File: package.json [hunk 2]
```diff
@@ -42,5 +42,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "prettier": "2.0.5"
   }
```
Comment (sidvenu) [inline package.json]: Can we move this out of this PR? I've just created an issue - #34 for using prettier. Reason being we may need to format code that is outside the scope of this PR as well.

File: src/components/Navbar.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            showDropdown: false,
+        }
+        this.change = this.change.bind(this)
+    }
+    componentDidMount() {
+        // if(localStorage.getItem('token') !== null){
+            this.props.getInfo()
+            this.props.getUser()
+        // }
+
```
Comment (sidvenu) [inline src/components/Navbar.js]: Comment to be removed?

**Review by sidvenu — CHANGES_REQUESTED** [review]
Also, please add new lines to the files you've created in the current PR that doesn't have a new line yet.

**Review by abha224 — APPROVED** [review]

File: src/components/Register.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        this.onChange = this.onChange.bind(this);
+    }
+
+    handleShow = (e) => this.setState({
+        showPassword: !this.state.showPassword
+    })
+
+    handleShow2 = (e) => this.setState({
+        showConfirmPassword: !this.state.showConfirmPassword
+    })
```
Comment (sidvenu) [inline src/components/Register.js]: Can we have better variable names for these two?

File: src/index.js [hunk 5]
```diff
@@ -15,48 +13,3 @@ ReactDOM.render (
...
-//       </Provider>
-//     )
-//   }
-// }
-
-
-
-// If you want your app to work offline and load faster, you can change
-// unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
```
Comment (sidvenu) [inline src/index.js]: Doesn't belong here.

**Review by sidvenu — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 5 above)
Comment (bismitaguha) [reply src/index.js]: These were unnecessary comments which I had added, so I removed them!

Comment (sidvenu) [reply src/index.js]: True yes, it does make sense to make this change - but can you shift this to a different PR? This does not belong to this issue, nor does it belong to any other issues you made this change in. I'll create an issue for it, if it helps?

**Review by sidvenu — APPROVED** [review]

**Comment by abha224** [comment]
@bismitaguha can you resolve the conflicts?

**Comment by bismitaguha** [comment]
@abha224 Merge conflicts resolved!
I feel such conflicts will arise in further PRs too.
